File: 1768. Merge Strings Alternately/Solution.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        _min = min(len(word1),len(word2))
        i = 0
        ans = []
        while i<_min:
            ans += word1[i]
            ans += word2[i]
            i += 1

        if(i<len(word1)):
            ans += word1[i:]
        else:
            ans += word2[i:]


        return "".join(ans)

File: 1768. Merge Strings Alternately/Solution.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int min = Math.min(word1.length(),word2.length());

        int i = 0;
        StringBuilder ans = new StringBuilder();

        while(i<min){
            ans.append(word1.charAt(i));
            ans.append(word2.charAt(i));
            i++;
        }

        if(i<word1.length()){
            ans.append(word1.toCharArray(),i,word1.length()-i);
        }else{
            ans.append(word2.toCharArray(),i,word2.length()-i);
        }

        return ans.toString();
    }

    public static void main(String[] args) {
        System.out.println(new Solution().mergeAlternately("abc", "pqr"));
    }
}